File: src/backend/scrape.py
from bs4 import BeautifulSoup as bs
import requests

content = requests.get("https://www.caringseniorservice.com/blog/senior-discount-guides-where-to-save-money-at-50-to-65").text
content

soup = bs(content, 'html.parser')

bodySoup = soup.find('div', {'class': 'ucBlogPostBody'}).find('article')
print(bodySoup.find_all('li'))
for li in bodySoup.find_all('li'):
  name=li.find('strong').text
  discount=li.text.split(': ')[-1].split('(')[0]
  age=li.text.split("(")[-1] if len(li.text.split("(")) > 1 else ''


  if '+' in age:
    age = int(age[:-2])
  else:
    age = None
    
  print(name.strip(':'), '|', discount, '|', age)w

File: src/backend/scrape.js
import firebase from './firebase.js';

import axios from 'axios';
import cheerio from 'cheerio';

axios.get("https://www.caringseniorservice.com/blog/senior-discount-guides-where-to-save-money-at-50-to-65")
  .then(response => {
    const content = response.data;
    const $ = cheerio.load(content);
    const bodySoup = $('.ucBlogPostBody').find('article');
    bodySoup.find('li').each((index, element) => {
      const li = $(element);

      const name = li.find('strong').text();
      let discount = li.text().split(': ')[1];
      if (discount) {
        discount = discount.split('(')[0];
      }
      let age = li.text().split("(")[1] || '';
      
      if (age.includes('+')) {
        age = parseInt(age.slice(0, -2));
      } else {
        age = null;
      }
      
      console.log(name.trim(':'), '|', discount, '|', age);
    });
  })
  .catch(error => {
    console.log(error);
  });



